scipy/stats/_stats_py.py

```diff
@@ -10944,6 +10947,161 @@ def linregress(x, y=None, alternative='two-sided'):
                             intercept_stderr=intercept_stderr)
 
 
+def _xp_mean(x, /, *, axis=None, weights=None, keepdims=False, nan_policy='propagate',
+             dtype=None, xp=None):
+    r"""Compute the arithmetic mean along the specified axis.
+
+    Parameters
+    ----------
+    x : real floating array
```

Comment:
Perhaps misleading given the casting of integers?
```suggestion
    x : real array
```